storybookjs/storybook
# Add previewMainTemplate option to StorybookConfig interface

Closes #21579 



## What I did

Updated the `StorybookConfig` interface to include the `previewMainTemplate` option because otherwise it would give a typescript error if users tried to include this valid config option.

## How to test

1. Modify `main.ts` in `ui/.storybook` to include the `previewMainTemplate` option. [The default one here could work as an example value for this option.](https://github.com/storybookjs/storybook/blob/2973677d0c32669465627c0480a3c1e7709efdec/code/lib/builder-webpack5/templates/preview.ejs)
2. Adding this option should not give a typescript error.
3. Start the storybook dev server
4. All stories should load properly within the new preview template.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
